### PR TITLE
add attributes user docs

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add the user docs for table/partition attributes

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: attributes.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+---
+title: 表属性及分区属性使用文档
+summary: 介绍 TiDB 的 `ATTRIBTUES` 使用方法。
```
Comment (en-jin19) [inline attributes.md:3]:
```suggestion
summary: 介绍 TiDB 的 `ATTRIBUTES` 使用方法。
```

**Comment by TomShawn** [comment]
@rleungx 

- 能否改一个文件名？一个单词的文件名不容易区分，能否更具体一点，比如 `table-partition-attributes.md`？
- 这篇文档需要添加至 TOC.md 目录文件。https://github.com/pingcap/docs-cn/blob/master/TOC.md 放在目录中哪个位置较为合适，是否有好的建议？
- 这篇文档没有叙述这个功能的【用户使用场景】，在什么场景下需要使用到这个功能，能帮助用户解决什么问题或带来什么帮助？

File: table-attributes.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
+---
+title: 表属性使用文档
+summary: 介绍 TiDB 的 `ATTRIBUTES` 使用方法。
+---
+
+表属性属性是从 v5.3.0 引入的新特性，主要用于为表或分区添加特定的属性以执行相应的属性对应的行为。
```
Comment (TomShawn) [inline table-attributes.md:6]: 能具体说一下【以执行相应的属性对应的行为】吗？这样的表述比较含糊不清。

Comment (rleungx) [reply table-attributes.md:6]: 这里是一个可扩展的特性，但目前只支持了 merge，所以如果后续有不同的属性，可能具体行为也不一样

File: table-attributes.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+```sql
+alter table t partition p attributes[=]'merge_option=deny';
+```
+
+### 允许属于某个分区的 Region 被合并
+```sql
+alter table t partition p attributes[=]'merge_option=allow';
+```
+
+### 重置某个分区的属性
```
Comment (TomShawn) [inline table-attributes.md:85]: 这个和上面的【## 重置表或分区属性】有什么区别呢？从 SQL 上看似乎差不多

File: table-attributes.md (hunk 2 above)
Comment (TomShawn) [reply table-attributes.md:6]:
```suggestion
表属性是从 v5.3.0 引入的新特性，用于为表或分区添加特定的属性，以对表或分区执行相应属性对应的操作，例如可以利用表属性控制 Region 的合并。
```

File: table-attributes.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
+---
+title: 表属性使用文档
+summary: 介绍 TiDB 的 `ATTRIBUTES` 使用方法。
+---
+
```
Comment (TomShawn) [inline table-attributes.md:5]:
```suggestion

# 表属性使用文档

```

File: table-attributes.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+表属性属性是从 v5.3.0 引入的新特性，主要用于为表或分区添加特定的属性以执行相应的属性对应的行为。
+
+> **注意：** 
+> 
+> 目前仅支持添加 `merge_option` 属性。
+
+# 使用方法
```
Comment (TomShawn) [inline table-attributes.md:12]:
```suggestion
## 基本使用方法
```

File: table-attributes.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+表属性为 `key=value` 的形式，多个属性需要用逗号分隔。具体示例如下，其中 `t` 为所要修改的表名，`p` 为所要修改的分区名，`[]`内部为可选项。
+
+## 设置表或分区属性
```
Comment (TomShawn) [inline table-attributes.md:16]: 建议将具体使用方法的标题都改为 ### 三级标题，并在标题与 ``` 之间留一个空格。

File: table-attributes.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+当表和分区同时存在相同属性的情况下，分区会将表的属性覆盖，如对于分区表 `t` 中分区 `p` 配置下述两个属性后：
+
+```sql
+alter table t attributes[=]'key=value';
+alter table t partition p attributes[=]'key=value1';
+```
+
+则分区 `p` 实际生效的属性为 `key=value1`。
+
+# 使用表属性控制 Region 合并
```
Comment (TomShawn) [inline table-attributes.md:51]:
```suggestion
## 使用表属性控制 Region 合并
```

File: table-attributes.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+## 使用场景描述
```
Comment (TomShawn) [inline table-attributes.md:53]:
```suggestion
### 使用场景描述
```

File: table-attributes.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
++ 场景一：在对某张新建表或某个新建分区写入数据存在热点问题时，通常会使用分裂打散 Region 的操作避免写入热点，但由于新建表或分区的分裂操作实际产生的是空 Region，如果分裂打散操作距离写入存在一定时间间隔，则 Region 会被合并，从而导致无法真正规避写入热点问题。可通过为表或分区添加 `merge_option` 属性，设置值为 `deny` 解决。
```
Comment (TomShawn) [inline table-attributes.md:55]:
```suggestion
+ 场景一：在对某张新建表或某个新建分区写入数据存在热点问题时，通常需要用户使用分裂打散 Region 的操作避免写入热点。但由于新建表或分区的分裂操作实际产生的是空 Region，如果分裂打散操作距离写入存在一定时间间隔，则 Region 会被合并，从而导致无法真正规避写入热点问题。该场景下，可通过为表或分区添加 `merge_option` 属性，设置值为 `deny`，有效解决写入热点问题。
```

File: table-attributes.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
++ 场景二：只读场景下，如果是通过手动分裂 Region 缓解某个表或分区的周期性读热点问题，且不希望热点消失后，手动分裂的 Region 被合并，则也可以通过为表或分区添加 `merge_option` 属性，设置值为 `deny` 解决。
```
Comment (TomShawn) [inline table-attributes.md:56]:
```suggestion
+ 场景二：在只读场景下，如果是通过手动分裂 Region 缓解某张表或分区的周期性读热点问题，且不希望热点消失后手动分裂的 Region 被合并，可以通过为表或分区添加 `merge_option` 属性，设置值为 `deny` 解决这一问题。
```

File: table-attributes.md (hunk 3 above)
Comment (rleungx) [reply table-attributes.md:85]: 上面的是通用语法，下面是针对 merge_option 的

File: table-attributes.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+```sql
+select * from information_schema.attributes where id='schema/t[/p]';
+```
+
+### 查看拥有某属性的所有表及分区
+
+```sql
+select * from information_schema.attributes where attributes like '%key%';
+```
```
Comment (TomShawn) [inline table-attributes.md:46]:
```suggestion
+ 设置表或分区属性

    ```sql
    alter table t [partition p ]attributes[=]'key=value[, key1=value1...]';
    ```

+ 重置表或分区属性

    ```sql
    alter table t [partition p ]attributes[=]default;
    ```

+ 查看当前全部表及分区属性

    ```sql
    select * from information_schema.attributes;
    ```

+ 查看某一张表或分区配置的属性

    ```sql
    select * from information_schema.attributes where id='schema/t[/p]';
    ```

+ 查看拥有某属性的所有表及分区

    ```sql
    select * from information_schema.attributes where attributes like '%key%';
    ```
```

File: table-attributes.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+### 覆盖关系
```
Comment (TomShawn) [inline table-attributes.md:48]:
```suggestion
### 覆盖关系

```

File: table-attributes.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+当表和分区同时存在相同属性的情况下，分区会将表的属性覆盖，如对于分区表 `t` 中分区 `p` 配置下述两个属性后：
```
Comment (TomShawn) [inline table-attributes.md:49]:
```suggestion
如果表和分区同时存在相同属性，分区会将表的属性覆盖。例如，为分区表 `t` 中分区 `p` 同时配置 `key-value` 和 `key-value1` 两个属性：
```

File: table-attributes.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+```sql
+alter table t attributes[=]'merge_option=deny';
+alter table t partition p attributes[=]'merge_option=allow';
+```
+
+当对分区表 `t` 中的分区 `p` 同时配置上述两个属性时，实际分区 `p` 的 Region 可以被合并。当分区的属性被重置时，分区 `p` 则会继承表 `t` 的属性，Region 无法被合并。
+
+> **注意：** 
+> 
+> 1. 如果目前只存在分区表的属性，即使配置 `merge_option=allow`，分区也会默认按照实际分区数量切分成多个 Region。如需合并所有 Region，则需要[重置该分区表](#使用方法)的属性。
```
Comment (en-jin19) [inline table-attributes.md:114]:
```suggestion
> 1. 如果目前只存在分区表的属性，即使配置 `merge_option=allow`，分区也会默认按照实际分区数量切分成多个 Region。如需合并所有 Region，则需要[重置该分区表的属性](#使用方法)。
```
```suggestion
> 1. 如果目前只存在分区表的属性，即使配置 `merge_option=allow`，分区也会默认按照实际分区数量切分成多个 Region。如需合并所有 Region，则需要[重置该分区表](#使用方法)的属性。
```

File: table-attributes.md [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+---
+
+# 表属性使用文档
+
+表属性是 TiDB 从 5.3.0 版本开始引入的新特性，用于为表或分区添加特定的属性，以对表或分区执行相应属性对应的操作，例如可以利用表属性控制 Region 的合并。
+
+> **注意：** 
+> 
+> 1. 目前仅支持添加 `merge_option` 属性。
+> 2. 在使用 TiDB Binlog 或 TiCDC 进行同步以及使用 BR 进行增量备份的时候会跳过该 DDL，如需使用，需要手动在下游或者备份集群执行。
```
Comment (en-jin19) [inline table-attributes.md:13]:
```suggestion
> 2. 在使用 TiDB Binlog 或 TiCDC 进行同步以及使用 BR 进行增量备份的时候会跳过该 DDL。如需使用，需要手动在下游或者备份集群执行。
```

**Review by nolouch — APPROVED** [review]

**Review by en-jin19 — APPROVED** [review]
LGTM

File: table-attributes.md [hunk 16]
```diff
@@ -0,0 +1,116 @@
+---
+title: 表属性使用文档
+summary: 介绍 TiDB 的 `ATTRIBUTES` 使用方法。
+---
+
+# 表属性使用文档
+
+表属性是 TiDB 从 5.3.0 版本开始引入的新特性，用于为表或分区添加特定的属性，以对表或分区执行相应属性对应的操作，例如可以利用表属性控制 Region 的合并。
```
Comment (nolouch) [inline table-attributes.md:8]: 可以说明表属性主要影响的是 engine 数据的物理属性?

Comment (rleungx) [reply table-attributes.md:8]: 这个概念我们似乎没有明确的定义？

File: table-attributes.md [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+    select * from information_schema.attributes where id='schema/t[/p]';
+    ```
+
++ 查看拥有某属性的所有表及分区
+
+    ```sql
+    select * from information_schema.attributes where attributes like '%key%';
+    ```
+
+## 覆盖关系
```
Comment (hfxsd) [inline table-attributes.md:49]:
```suggestion
## 覆盖关系

```

File: table-attributes.md [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+```
+
+分区 `p` 实际生效的属性为 `key=value1`。
+
+## 使用表属性控制 Region 合并
+
+### 使用场景
+
++ 场景一：在对某张新建表或某个新建分区写入数据存在热点问题时，通常需要用户使用分裂打散 Region 的操作避免写入热点，但由于新建表或分区的分裂操作实际产生的是空 Region，如果分裂打散操作距离写入存在一定时间间隔，则 Region 会被合并，从而导致无法真正规避写入热点问题。该场景下，可通过为表或分区添加 `merge_option` 属性，设置值为 `deny`，有效解决写入热点问题。
++ 场景二：在只读场景下，如果是通过手动分裂 Region 缓解某张表或分区的周期性读热点问题，且不希望热点消失后手动分裂的 Region 被合并，可以通过为表或分区添加 `merge_option` 属性，设置值为 `deny` 解决这一问题。
```
Comment (hfxsd) [inline table-attributes.md:64]:
```suggestion
有时会遇到读或写入热点的问题。为解决该问题，你可以使用表属性控制 Region 合并，通过为表或分区添加 `merge_option` 属性，将其设置为 `deny`来解决。以下介绍了两种使用场景。

#### 新建表或分区的写入热点问题

在对某张新建表或某个新建分区写入数据存在热点问题时，通常需要使用分裂打散 Region 的操作避免写入热点，但由于新建表或分区的分裂操作实际产生的是空 Region，如果分裂打散操作距离写入存在一定时间间隔，则 Region 会被合并，从而导致无法真正规避写入热点问题。此时可以为表或分区添加 `merge_option` 属性，设置为 `deny` 来解决问题。

#### 只读场景下周期性读热点问题

在只读场景下，如果是通过手动分裂 Region 缓解某张表或分区的周期性读热点问题，且不希望热点消失后手动分裂的 Region 被合并。此时可以为表或分区添加 `merge_option` 属性，设置为 `deny` 来解决问题。
```

File: table-attributes.md [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+alter table t attributes[=]'merge_option=deny';
+alter table t partition p attributes[=]'merge_option=allow';
+```
+
+同时配置上述两个属性时，实际分区 `p` 的 Region 可以被合并。当分区的属性被重置时，分区 `p` 则会继承表 `t` 的属性，Region 无法被合并。
+
+> **注意：** 
+> 
+> 1. 如果目前只存在分区表的属性，即使配置 `merge_option=allow`，分区也会默认按照实际分区数量切分成多个 Region。如需合并所有 Region，则需要[重置该分区表的属性](#使用方法)。
+> 2. 使用该属性需要注意 PD 的参数 [`split-merge-interval`](/pd-configuration-file.md#split-merge-interval) 的配置。如果没有配置 `merge_option`，Region 在超过 `split-merge-interval` 指定的时间后满足条件即可合并。如果配置了 `merge_option`，则超过指定时间后会根据 `merge_option` 的配置情况再决定是否可以合并。
```
Comment (hfxsd) [inline table-attributes.md:116]:
```suggestion
> - 如果目前只存在分区表的属性，即使配置 `merge_option=allow`，分区也会默认按照实际分区数量切分成多个 Region。如需合并所有 Region，则需要[重置该分区表的属性](#使用方法)。
> - 使用该属性需要注意 PD 的参数 [`split-merge-interval`](/pd-configuration-file.md#split-merge-interval) 的配置。如果没有配置 `merge_option`，Region 在超过 `split-merge-interval` 指定的时间后满足条件即可合并。如果配置了 `merge_option`，则超过指定时间后会根据 `merge_option` 的配置情况再决定是否可以合并。
```

File: table-attributes.md [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+---
+
+# 表属性使用文档
+
+表属性是 TiDB 从 5.3.0 版本开始引入的新特性，用于为表或分区添加特定的属性，以对表或分区执行相应属性对应的操作，例如可以利用表属性控制 Region 的合并。
+
+> **注意：** 
+> 
+> 1. 目前仅支持添加 `merge_option` 属性。
+> 2. 在使用 TiDB Binlog 或 TiCDC 进行同步以及使用 BR 进行增量备份的时候会跳过该 DDL。如需使用，需要手动在下游或者备份集群执行。
```
Comment (qiancai) [inline table-attributes.md:13]:
```suggestion
> - 目前 TiDB 仅支持为表或分区添加 `merge_option` 属性。
> - 当使用 TiDB Binlog 或 TiCDC 进行同步或者使用 BR 进行增量备份时，同步和备份会跳过表属性设置 DDL。如需在下游或者备份集群使用表属性，需要在下游或者备份集群手动执行表属性设置 DDL。
```

File: table-attributes.md [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+    alter table t [partition p ]attributes[=]'key=value[, key1=value1...]';
+    ```
+
++ 重置表或分区属性
+
+    ```sql
+    alter table t [partition p ]attributes[=]default;
+    ```
+
++ 查看当前全部表及分区属性
```
Comment (qiancai) [inline table-attributes.md:31]:
```suggestion
+ 查看全部表及分区属性
```

File: table-attributes.md [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+    ```
+
++ 查看拥有某属性的所有表及分区
+
+    ```sql
+    select * from information_schema.attributes where attributes like '%key%';
+    ```
+
+## 覆盖关系
+如果为分区表配置属性，该属性则会对表的所有分区生效，如额外为分区配置属性且属性与表的属性相同时，分区会将表的属性覆盖。例如，当分区表 `t` 配置属性 `key=value`，同时分区 `p` 配置属性 `key=value1` 时：
```
Comment (qiancai) [inline table-attributes.md:50]:
```suggestion
为分区表配置的属性会对表的所有分区生效。一种例外情况是，如果分区表和分区都配置了相同属性但属性值不同，分区属性将覆盖分区表属性。例如，当分区表 `t` 配置属性 `key=value`，同时分区 `p` 配置属性 `key=value1` 时：
```

File: table-attributes.md [hunk 23]
```diff
@@ -0,0 +1,116 @@
+---
+title: 表属性使用文档
+summary: 介绍 TiDB 的 `ATTRIBUTES` 使用方法。
+---
+
+# 表属性使用文档
```
Comment (qiancai) [inline table-attributes.md:6]:
```suggestion
# 表属性
```

File: table-attributes.md [hunk 24]
```diff
@@ -0,0 +1,116 @@
+---
+title: 表属性使用文档
```
Comment (qiancai) [inline table-attributes.md:2]:
```suggestion
title: 表属性
```

File: table-attributes.md [hunk 25]
```diff
@@ -0,0 +1,124 @@
...
+---
+
+# 表属性
+
+表属性是 TiDB 从 5.3.0 版本开始引入的新特性，用于为表或分区添加特定的属性，以对表或分区执行相应属性对应的操作，例如可以利用表属性控制 Region 的合并。
+
+> **注意：** 
+> 
+> - 目前 TiDB 仅支持为表或分区添加 `merge_option` 属性，用于控制 Region 合并。
+> - 当使用 TiDB Binlog 或 TiCDC 进行同步或者使用 BR 进行增量备份时，同步和备份会跳过表属性设置的 DDL。如需在下游或者备份集群使用表属性，需要在下游或者备份集群手动执行表属性设置的 DDL。
```
Comment (TomShawn) [inline table-attributes.md:13]: @rleungx PTAL whether my comment is correct. Thanks!
```suggestion
> - 当使用 TiDB Binlog 或 TiCDC 进行同步或者使用 BR 进行增量备份时，同步和备份会跳过设置表属性的 DDL 语句。如需在下游或者备份集群使用表属性，需要在下游或者备份集群手动执行该 DDL 语句以设置表属性。
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

**Pull request merged**